src/client.js

```diff
@@ -19,65 +19,121 @@ class TouchPortalClient extends EventEmitter {
     this.customStates = {};
   }
 
-  createState(id, desc, defaultValue, parentGroup) {
-    if (this.customStates[id]) {
-      this.logIt('ERROR', `createState: Custom state of ${id} already created`);
-      throw new Error(`createState: Custom state of ${id} already created`);
-    }
-    this.customStates[id] = desc;
-    const newState = {
-      type: 'createState',
-      id: `${id}`,
-      desc: `${desc}`,
-      defaultValue: `${defaultValue}`,
-    };
-    if (parentGroup !== '' || parentGroup !== undefined) {
-      newState.parentGroup = `${parentGroup}`;
-    }
-    this.send(newState);
-  }
-
-  createStateMany(states) {
-    const createStateArray = [];
-
-    if (states.length <= 0) {
-      this.logIt('ERROR', 'createStateMany : states contains no data');
-      throw new Error('createStateMany: states contains no data');
+  /**
+   * Internal helper method for states
+   * 
+   * @param {Object[Object]} states - Array of object containing id,
+   * desc, defaultValue, optionally parentGroup.
+   * @param {String} stateType - TouchPortal state type removeState
+   * or createState.
+   * 
+   * @return {Array<Object>}
+   */
+  stateHelper(states, stateType) {
+    const stateArray = [];
+    if (!(typeof states == "object" || states.length)) {
+      this.logIt('ERROR', `${stateType} has to be vaild object.`)
     }
-
     states.forEach((state) => {
-      if (this.customStates[state.id]) {
-        this.logIt('WARN', `createState: Custom state of ${state.id} already created`);
-      } else {
-        this.customStates[state.id] = state.desc;
-        const newState = {
-          type: 'createState',
-          id: `${state.id}`,
-          desc: `${state.desc}`,
-          defaultValue: `${state.defaultValue}`,
+      const stateObject = {
```

Comment:
So... don't hate me but...  :)

What is the point of copying the object properties from the existing `state` instance to a new object?

You could just as well add the `stateType` to `state` and send that... no?

For the `remove*()` versions the `state` object only has the ID anyway, so no problem.

And, I know it was already like that in places, but code like ``${stateType}`` is redundant...  they're already strings, there's no reason to wrap them in another interpolated string.

Reply:
I just did it to followed how he did other part. but yea Ill remove it